server/src/test/java/org/opensearch/cluster/node/DiscoveryNodeTests.java

```diff
@@ -249,4 +257,70 @@ public void testDiscoveryNodeIsSearchNode() {
         final DiscoveryNode node = DiscoveryNode.createLocal(settingWithSearchRole, buildNewFakeTransportAddress(), "node");
         assertThat(node.isSearchNode(), equalTo(true));
     }
+
+    public void testToXContentInAPIMode() throws IOException {
+        final DiscoveryNode node = DiscoveryNode.createLocal(
+            Settings.EMPTY,
+            new TransportAddress(TransportAddress.META_ADDRESS, 9200),
+            "node_1"
+        );
+        XContentBuilder builder = JsonXContent.contentBuilder().prettyPrint();
+        builder.startObject();
+        node.toXContent(builder, new ToXContent.MapParams(singletonMap(Metadata.CONTEXT_MODE_PARAM, CONTEXT_MODE_API)));
+        builder.endObject();
+
+        String expectedNodeAPUXContent = "{\n"
+            + "  \"node_1\" : {\n"
+            + "    \"name\" : \""
+            + node.getName()
+            + "\",\n"
+            + "    \"ephemeral_id\" : \""
+            + node.getEphemeralId()
+            + "\",\n"
+            + "    \"transport_address\" : \"0.0.0.0:9200\",\n"
+            + "    \"attributes\" : { }\n"
+            + "  }\n"
+            + "}";
+
+        assertEquals(expectedNodeAPUXContent, builder.toString());
+    }
+
+    public void testToXContentInGatewayMode() throws IOException {
+        final DiscoveryNode node = DiscoveryNode.createLocal(
+            Settings.EMPTY,
+            new TransportAddress(TransportAddress.META_ADDRESS, 9200),
+            "node_1"
+        );
+        XContentBuilder builder = JsonXContent.contentBuilder().prettyPrint();
+        builder.startObject();
+        node.toXContent(builder, new ToXContent.MapParams(singletonMap(Metadata.CONTEXT_MODE_PARAM, CONTEXT_MODE_GATEWAY)));
+        builder.endObject();
+
+        String expectedNodeAPUXContent = "{\n"
```

Comment:
If you really want to test against ToXContentFragment interface you need to test partial JSON that doesn't start and end with `{` and `}`.  (You might find that your tests currently fail!)  

Or require start and end and implement `ToXContentObject` and this test is OK.

Reply:
Actually the start and end `{` `}` are coming from line 295, 297. To build a valid json, we need to have start and end objects in our builder or it will fail to create a Json string. As the start and end of json are coming from line 295 and 297, our toXContent is only returning the partial XContent fragment which is getting verified in our test.